### PR TITLE
Update reporting api dependency

### PR DESCRIPTION
**Pull request by flora-hofmann-frequenz**
Addressing issue #41

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -6,11 +6,13 @@
...
 
-<!-- Here goes notes on how to upgrade from previous versions, including deprecations and what they should be replaced with -->
+* Upgrading "frequenz-api-reporting >= 0.1.1, < 0.2" to "frequenz-api-reporting >= 0.2.0, < 0.3"
+* Upgrading "frequenz-client-common == 0.1.0" to "frequenz-client-common >= 0.2.0, < 0.3"
 
 ## New Features
 
-<!-- Here goes the main new features and examples or instructions on how to use them -->
+* The `ReportingApiClient` exposes the resampling option of the data, with its 
+resolution represented in second and its default set to `None`.
```
Comment (cwasicki) [inline RELEASE_NOTES.md:15]: This is not related to this update, should it be here? If so, better make this a separate commit.

File: docs/_scripts/macros.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def _slugify(text: str) -> str:
     # The type of the return value is not defined for the markdown library.
     # Also for some reason `mypy` thinks the `toc` module doesn't have a
     # `slugify_unicode` function, but it definitely does.
-    return toc.slugify_unicode(text, "-")  # type: ignore[attr-defined,no-any-return]
+    return toc.slugify_unicode(text, "-")
```
Comment (cwasicki) [inline docs/_scripts/macros.py:33]: Also a separate commit.

File: pyproject.toml
Comment (cwasicki) [inline pyproject.toml:1]: Are all of these updates in this file needed for the reporting API update?

File: RELEASE_NOTES.md (hunk 1 above)
Comment (flora-hofmann-frequenz) [reply RELEASE_NOTES.md:15]: I thought so too, but then it does not appear in our merged code base which is weird?
I had separate commits but squashed them - will add them again.

File: pyproject.toml
Comment (flora-hofmann-frequenz) [reply pyproject.toml:1]: Yes. The toml was quite outdated/didn't have all the dependencies needed and caused problems with the dependencies (I discussed this with Luca [here](https://frequenz-io.slack.com/archives/CNMCCUB5H/p1715934534748859)). All these updates are needed for the reporting API dependency update (that's also why the dependabot update failed).

**Review by cwasicki — APPROVED** [review]
